frontend/viewer/src/lib/components/stomp/stomp-safe-lcm-rich-text-editor.svelte

```diff
@@ -1,21 +1,33 @@
 <script lang="ts">
-  import type { ComponentProps } from 'svelte';
+  import {untrack, type ComponentProps} from 'svelte';
   import {StompGuard} from './stomp-guard.svelte';
   import {mergeProps} from 'bits-ui';
   import LcmRichTextEditor from '../lcm-rich-text-editor/lcm-rich-text-editor.svelte';
+  import {useIdleService} from '$lib/services/idle-service';
 
-  type Props = ComponentProps<typeof LcmRichTextEditor>;
+  type Props = ComponentProps<typeof LcmRichTextEditor> & { onchange: () => void };
 
-  let { value = $bindable(), ...rest}: Props = $props();
+  let { value = $bindable(), onchange, ...rest}: Props = $props();
 
   const guard = new StompGuard(
     () => value,
     (newValue) => value = newValue
   );
+
+  let idleService = useIdleService();
+  function onIdle() {
+    if (guard.isDirty) {
+      onchange();
```

Comment:
I think we also need to do `guard.commitAndUnlock` otherwise it'll stay locked? Not sure though

Reply:
Yes, I think you're right. I'm not sure how I would artificially create a situation where the input would get stomped (I think I'd have to get two copies of FW Lite running on the same project to set that situation up), but judging by the code alone, I believe the `onchange` that is being called here would *not* also trigger the input's `onchange` handler that calls `commitAndUnlock`. And if we've saved a CRDT change, we want that to be the new protected-from-stomping value. Will fix.

Reply:
Fixed in commit b60a59bf2.